src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -78,6 +81,67 @@ CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::Write(const ConcreteDataAttribu
 
 } // namespace
 
+namespace chip {
+
+// -----------------------------------------------------------------------------
+// OtaRequestorServer implementation
+
+static OtaRequestorServer sInstance;
```

Comment:
Do we need another global object for that? Maybe we could maintain the state in `OTARequestor` object and use `OtaSoftwareUpdateRequestorAttrAccess` above to return the state from the `OTARequestor` object via `OTARequestorInterface`? That would be more consistent, the state would be less scattered across different objects and applications could use `OTARequestorInterface` to easily access the state to (e.g. to display the progress bar or so). What do you think?

Reply:
Spoke offline. Will keep the update as is but will remove this global instance and just make these Set functions global.